lib/trollop.rb

```diff
@@ -76,8 +77,18 @@ def initialize(*a, &b)
     @synopsis = nil
     @usage = nil
     
+    ## allow passing settings through Parser.new as an optional hash.
+    ## but keep compatibility with non-hashy args, though.
+    begin
+      @settings = Hash[*a]
+      a=[] ## clear out args if using as settings-hash
+    rescue ArgumentError
+      @settings = Hash.new()
+    end
+
     # instance_eval(&b) if b # can't take arguments
-    cloaker(&b).bind(self).call(*a) if b
+    #cloaker(&b).bind(self).call(*a) if b
+    self.instance_exec(*a,&b) if block_given?
```

Comment:
what is cloaker?
Is this a __why fix for a ruby 1.8.6 limitation?

Reply:
We've been using methods as a way to introduce "settings"
Do we really want to change this?

stylistically, it is nice.

Reply:
The cloaker is the _why fix.  Unfortunately it doesn't seem to work properly with more complicated arg passing.  It has been months since i've last looked at that.

Using params for setting "global" settings on `Trollop.options` seemed to me a better than adding them to as methods, as the methods are sequenced in an order.  If you have a method named `disable_auto_short_options`, then you'd need to set that before any `opt` method call.  this would either require running the "settings" methods first to get a sane result.